_books/ion-1-1/src/modules/encoding_modules.md

```diff
@@ -0,0 +1,228 @@
+# Encoding modules
+
+The encoding of each segment of a stream is shaped by the currently configured _encoding modules_,
+an ordered sequence of modules that determine which symbols and macros are available for use in the stream.
+A writer can modify this sequence by emitting an [encoding directive](directives.md#encoding-directives).
+
+By logically concatenating the encoding modules' symbol and macro tables respectively,
+they can be viewed as unified local symbol and macro tables.
+
+For example, consider this encoding directive:
+```ion
+$ion::
+(encoding
+    (module mod_a
+        (symbol_table ["a", "b", "c"])
+        (macro_table
+            (macro foo () Foo)
+            (macro bar () Bar)))
+    (module mod_b
+        (symbol_table ["c", "d", "e"])
+        (macro_table
+            (macro baz () Baz)
+            (macro quux () Quux)))
+    (module mod_c
+        (symbol_table ["f", "g", "h"])
+        (macro_table
+            (macro quuz () Quuz)
+            (macro foo () Foo2)))))
+```
+It produces the encoding module sequence `mod_a mod_b mod_c`.
+
+The segment's local symbol table, formed by logically concatenating the symbol tables of `mod_a`,
+`mod_b`, and `mod_c` in that order, is:
+
+| Address | Symbol text |
+|:-------:|-------------|
+|   `0`   | `a`         |
+|   `1`   | `b`         |
+|   `2`   | `c`         |
+|   `3`   | `c`         |
+|   `4`   | `d`         |
+|   `5`   | `e`         |
+|   `6`   | `f`         |
+|   `7`   | `g`         |
+|   `8`   | `h`         |
+
+Notice that no de-duplication takes place; `c` appears in both addresses `3` and `4`.
+
+The segment's macro table, formed by logically concatenating the macro tables of `mod_a`,
+`mod_b`, and `mod_c` in that order, is:
+
+| Address | Macro         |
+|:-------:|---------------|
+|   `0`   | `mod_a::foo`  |
+|   `1`   | `mod_a::bar`  |
+|   `2`   | `mod_b::baz`  |
+|   `3`   | `mod_b::quux` |
+|   `4`   | `mod_c::quuz` |
+|   `5`   | `mod_c::foo`  |
+
+Notice that `mod_a::foo` and `mod_c::foo` can coexist in this unified view without issue.
+Invocations of these macros require that they be qualified by their enclosing module's name.
+
+Because lower addresses take fewer bytes to encode than higher addresses,
+writers should place the modules they anticipate referencing the most frequently at the beginning of the encoding module sequence.
+
+Modules in the current segment's encoding module sequence are said to be _active_,
+while modules that are defined or imported but which are not in the encoding module sequence are _available_.
+E-expressions can only invoke macros in an active module.
+
+For example:
+```ion
+$ion::
+(module mod_a
+    (macro_table
+        (macro foo () Foo)))
+
+// `mod_a` is now available
+
+$ion::
+(module mod_b
+    (macro_table
+        (macro bar () Bar)))
+
+// `mod_b` is now available
+
+$ion::
+(encoding mod_a)
```

Comment:
Does this remove `_` from the active modules?

Reply:
No; `_` is always implicitly at the head of the encoding module sequence. This is mentioned in the _encoding directive_ section, but I've also added it to this section as part of addressing an earlier comment.